lib/combine_pdf/parser.rb

```diff
@@ -243,22 +243,22 @@ def _parse_
         ##########################################
         elsif str = @scanner.scan(/\<[0-9a-fA-F]*\>/)
           # warn "Found a hex string"
-          str = str.slice(1..-2).force_encoding(Encoding::ASCII_8BIT)
+          str = str.slice(1..-2).dup.force_encoding(Encoding::ASCII_8BIT)
           # str = "0#{str}" if str.length.odd?
-          out << unify_string([str].pack('H*').force_encoding(Encoding::ASCII_8BIT))
+          out << unify_string([str].pack('H*').dup.force_encoding(Encoding::ASCII_8BIT))
```

Comment:
(again) This dup is redundant and will use up a bunch of memory. The dynamic String returned from `Array#pack` is mutable.

Reply:
Agreed.